docs/content/integrations/duckdb/reference.mdx

```diff
@@ -7,17 +7,56 @@ description: Store your Dagster assets in DuckDB
 
 This reference page provides information for working with [`dagster-duckdb`](/\_apidocs/libraries/dagster-duckdb) features that are not covered as part of the [Using Dagster with DuckDB tutorial](/integrations/duckdb/using-duckdb-with-dagster).
 
+DuckDB resource:
+
+- [Executing custom SQL queries](#executing-custom-sql-queries)
+
+DuckDB I/O manager:
+
 - [Selecting specific columns in a downstream asset](#selecting-specific-columns-in-a-downstream-asset)
 - [Storing partitioned assets](#storing-partitioned-assets)
 - [Storing tables in multiple schemas](#storing-tables-in-multiple-schemas)
 - [Using the DuckDB I/O manager with other I/O managers](#using-the-duckdb-io-manager-with-other-io-managers)
 - [Storing and loading PySpark or Polars DataFrames in DuckDB](#storing-and-loading-pyspark-or-polars-dataframes-in-duckdb)
 - [Storing multiple DataFrame types in DuckDB](#storing-multiple-dataframe-types-in-duckdb)
-- [Executing custom SQL commands with the DuckDB resource](#executing-custom-sql-commands-with-the-duckdb-resource)
 
 ---
 
-## Selecting specific columns in a downstream asset
+## DuckDB resource
+
+The DuckDB resource provides access to a [`duckdb.DuckDBPyConnection`](https://duckdb.org/docs/api/python/reference/#duckdb.DuckDBPyConnection) object. This allows you full control over how your data is stored and retrieved in your database.
+
+For further information on the DuckDB resource, see the [DuckDB resource API docs](/\_apidocs/libraries/dagster-duckdb#dagster_duckdb.DuckDBResource).
+
+### Executing custom SQL queries
+
+```python file=/integrations/duckdb/reference/resource.py startafter=start endbefore=end
+from dagster_duckdb import DuckDBResource
+
+from dagster import asset
+
+# this example executes a query against the iris_dataset table created in Step 2 of the
+# Using Dagster with DuckDB tutorial
+
+
+@asset(deps=[iris_dataset])
+def small_petals(duckdb: DuckDBResource) -> None:
+    with duckdb.get_connection() as conn:  # conn is a DuckDBPyConnection
+        conn.cursor().execute(
```

Comment:
Do we need the cursor to run a query? I feel like I've been directly calling `.execute` from the connection, but it's been a minute since I tried.

Reply:
good point - i'm actually not sure so i will test

Reply:
i'm getting majorly weird issues trying to run anything with duckdb - seems like the commands are just hanging forever with no output or errors... and it's happening with both `cursor()` and raw `execute()`. I think it's completely unrelated to the actual code snippet and is rather some strange issue with my duckdb setup. I'm going to timebox myself like 30 min to see if i can figure out what's up, and then go from there

Reply:
Oh, I've seen this before.

One of the reasons why is if you're making a new db file in a directory that doesn't exist, duckdb won't do the `mkdir -p` for you, and the Dagster step will hang forever.

Could that be it?

Reply:
oh maybe, i'll check that

Reply:
deleting the db and letting duckdb make it from scratch again did it. so something just got wonky somewhere. but the `cursor()` is not required! I'll remove it

Reply:
Oh, interesting. Do you think the duckdb versions can be different? Because DuckDB storage isn't compatible across patch versions, so I'm wondering if it's silently erroring or we're not catching the exception properly.